scipy/spatial/transform/_rotation.pyx

```diff
@@ -1107,13 +1117,40 @@ cdef class Rotation:
         # If a single matrix is given, convert it to 3D 1 x 3 x 3 matrix but
         # set self._single to True so that we can return appropriate objects in
         # the `to_...` methods
-        cdef double[:, :, :] cmatrix
         if matrix.shape == (3, 3):
-            cmatrix = matrix[None, :, :]
+            matrix = matrix[np.newaxis, :, :]
             is_single = True
-        else:
-            cmatrix = matrix
 
+        # Calculate the determinant of the rotation matrix
+        # (should be positive for right-handed rotations)
+        dets = np.linalg.det(matrix)
+        if np.any(dets <= 0):
+            ind = np.where(dets <= 0)[0][0]
+            raise ValueError("Non-positive determinant (left-handed or null "
+                             f"coordinate frame) in rotation matrix {ind}: "
+                             f"{matrix[ind]}.")
+
+        # Gramian orthonormality check
+        # (should be the identity matrix for orthonormal matrices)
+        # Note that we have already ruled out left-handed cases above
+        gramians = matrix @ np.swapaxes(matrix, 1, 2)
+        is_orthonormal = np.all(np.isclose(gramians, np.eye(3), atol=1e-12),
+                                axis=(1, 2))
```

Comment:
(minor) I think "orthogonal matrix" is more standard and well understood term, I don't know exact reasons, but "orthonormal" is rarely used as far as I'm aware.

Reply:
Orthogonal doesn't capture the unit norm quality. I could switch to unitary?

Reply:
We can use unitary here as we use Hermitian for symmetric real anyways. Unitary is also easier to link in your head to something being 1.0 though you can't immediately remember what it would be, slightly easier to type too. But not a blocker, I don't have strong opinions.

Reply:
> Orthogonal doesn't capture the unit norm quality.

Again I don't known deep reasons, but SO3 matrices are almost always called orthogonal (from my experience).

Reply:
> Orthogonal doesn't capture the unit norm quality. 

FYI: The definition of *orthogonal matrix* includes orthonormality.

Reply:
> The definition of orthogonal matrix includes orthonormality.

Whelp, I looked it up and looks like I wasn't remembering correctly. -2 linear algebra points for me :) 

Updated the wording since orthogonal is a more common term.

Reply:
It should be the other way around. `a = np.array([[1, 2], [-2, -1]])` is only orthogonal but not **ortho**gonal and **normal**ized. I know this is a convention and definition issue but I know many fields that orthogonal/orthonormal difference is a real thing.

Reply:
[Orthogonal matrix](https://en.wikipedia.org/wiki/Orthogonal_matrix) entry does not make a distinction but [Unitary Matrix](https://en.wikipedia.org/wiki/Unitary_matrix#Equivalent_conditions) does and mentions orthonormality. So I guess this is one of those "it depends who you talk to" issues.

Reply:
I'm also finding some sources that call any matrix with perpendicular basis vectors an orthogonal matrix. But others saying that the unit norm is an additional constraint. I don't think it matters too much, the user input is normalized and the code should be clear for anyone making future modifications.

Reply:
Indeed, and also in the rotation context it is pretty much given that rotation matrices are unitary, or should be.